[WTF] Transition non JSC and WebCore code to use a checked HashMap (3/3)

zakariaridouh opened the pull request:
#### bbf5bdae3c29f127d320f305742b33ffe112d390
<pre>
[WTF] Transition non JSC and WebCore code to use a checked HashMap (3/3)
<a href="https://bugs.webkit.org/show_bug.cgi?id=278488">https://bugs.webkit.org/show_bug.cgi?id=278488</a>
<a href="https://rdar.apple.com/134445131">rdar://134445131</a>

Reviewed by Geoffrey Garen.

This commit spreads the checked WTF::HashMap more across the code base,
avoiding the performance sensitive areas of the code like WebCore and
JavaScriptCore. Some parts of these areas were switched over, and
performance is being monitored. We continue using UncheckedKeyHashMap in
these regions of the code.

This commit landing is contingent on no power and performance
regressions.

* Source/JavaScriptCore/inspector/scripts/codegen/cpp_generator_templates.py:
* Source/WTF/wtf/RobinHoodHashMap.h:
* Source/WTF/wtf/persistence/PersistentCoders.h:
* Source/WTF/wtf/win/FileSystemWin.cpp:
(WTF::FileSystemImpl::cachedStorageDirectory):
* Source/WebCore/Modules/cache/DOMCacheEngine.cpp:
(WebCore::DOMCacheEngine::queryCacheMatch):
* Source/WebCore/Modules/cache/DOMCacheEngine.h:
* Source/WebCore/Modules/indexeddb/shared/IDBDatabaseInfo.cpp:
(WebCore::IDBDatabaseInfo::IDBDatabaseInfo):
* Source/WebCore/Modules/indexeddb/shared/IDBDatabaseInfo.h:
(WebCore::IDBDatabaseInfo::IDBDatabaseInfo):
(WebCore::IDBDatabaseInfo::objectStoreMap const):
* Source/WebCore/Modules/indexeddb/shared/IDBObjectStoreInfo.cpp:
(WebCore::IDBObjectStoreInfo::IDBObjectStoreInfo):
* Source/WebCore/Modules/indexeddb/shared/IDBObjectStoreInfo.h:
(WebCore::IDBObjectStoreInfo::IDBObjectStoreInfo):
(WebCore::IDBObjectStoreInfo::indexMap const):
* Source/WebCore/bindings/js/RunJavaScriptParameters.h:
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::setComposition):
* Source/WebCore/editing/Editor.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/editing/cocoa/AttributedString.mm:
(WebCore::AttributedString::AttributedString):
(WebCore::AttributedString::rangesAreSafe):
(WebCore::toNSDictionary):
(WebCore::extractDictionary):
* Source/WebCore/html/PermissionsPolicy.h:
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::setActiveContentRuleListActionPatterns):
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/ResourceLoadObserver.h:
(WebCore::ResourceLoadObserver::setDomainsWithCrossPageStorageAccess):
* Source/WebCore/page/DiagnosticLoggingClient.h:
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::updateStorageAccessUserAgentStringQuirks):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/platform/EventTrackingRegions.h:
* Source/WebCore/platform/ScreenProperties.h:
* Source/WebCore/platform/SerializedPlatformDataCueValue.h:
* Source/WebCore/platform/network/NetworkStorageSession.cpp:
(WebCore::NetworkStorageSession::setDomainsWithCrossPageStorageAccess):
(WebCore::NetworkStorageSession::grantStorageAccess):
(WebCore::NetworkStorageSession::storageAccessQuirks):
* Source/WebCore/platform/network/NetworkStorageSession.h:
* Source/WebCore/platform/network/OrganizationStorageAccessPromptQuirk.h:
(WebCore::OrganizationStorageAccessPromptQuirk::OrganizationStorageAccessPromptQuirk):
* Source/WebCore/platform/network/RFC8941.cpp:
(RFC8941::parseParameters):
(RFC8941::parseDictionary):
(RFC8941::parseDictionaryStructuredFieldValue):

ADDITIONAL FILES REDACTED FROM COMMIT MESSAGE FOR BREVITY

Canonical link: <a href="https://commits.webkit.org/285616@main">https://commits.webkit.org/285616@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5ede5830007efa19c22bf9459170368debeb095b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [⏳ 🧪 style ](https://ews-build.webkit.org/#/builders/Style-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52687 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26066 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/77487 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24496 "Built successfully") 
| [⏳ 🧪 bindings ](https://ews-build.webkit.org/#/builders/Bindings-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/60493 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/470 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/57570 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/16042 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [⏳ 🧪 webkitperl ](https://ews-build.webkit.org/#/builders/WebKitPerl-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47596 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63040 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/37989 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [⏳ 🧪 webkitpy ](https://ews-build.webkit.org/#/builders/WebKitPy-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/44235 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20519 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/22825 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/66393 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66093 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20872 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79140 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/72515 "Built successfully and passed tests") | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-2-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/65997 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [⏳ 🧪 services ](https://ews-build.webkit.org/#/builders/Services-EWS "Waiting in queue, processing has not started yet") | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/714 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63049 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65275 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16134 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7280 "Passed tests") | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/537 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/3274 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/20736 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/567 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/551 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/569 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren reviewed (approved):
r=me pending EWS and perf tests

zakariaridouh committed:
[WTF] Transition non JSC and WebCore code to use a checked HashMap (3/3)
https://bugs.webkit.org/show_bug.cgi?id=278488
rdar://134445131

Reviewed by Geoffrey Garen.

This commit spreads the checked WTF::HashMap more across the code base,
avoiding the performance sensitive areas of the code like WebCore and
JavaScriptCore. Some parts of these areas were switched over, and
performance is being monitored. We continue using UncheckedKeyHashMap in
these regions of the code.

This commit landing is contingent on no power and performance
regressions.

* Source/JavaScriptCore/inspector/scripts/codegen/cpp_generator_templates.py:
* Source/WTF/wtf/RobinHoodHashMap.h:
* Source/WTF/wtf/persistence/PersistentCoders.h:
* Source/WTF/wtf/win/FileSystemWin.cpp:
(WTF::FileSystemImpl::cachedStorageDirectory):
* Source/WebCore/Modules/cache/DOMCacheEngine.cpp:
(WebCore::DOMCacheEngine::queryCacheMatch):
* Source/WebCore/Modules/cache/DOMCacheEngine.h:
* Source/WebCore/Modules/indexeddb/shared/IDBDatabaseInfo.cpp:
(WebCore::IDBDatabaseInfo::IDBDatabaseInfo):
* Source/WebCore/Modules/indexeddb/shared/IDBDatabaseInfo.h:
(WebCore::IDBDatabaseInfo::IDBDatabaseInfo):
(WebCore::IDBDatabaseInfo::objectStoreMap const):
* Source/WebCore/Modules/indexeddb/shared/IDBObjectStoreInfo.cpp:
(WebCore::IDBObjectStoreInfo::IDBObjectStoreInfo):
* Source/WebCore/Modules/indexeddb/shared/IDBObjectStoreInfo.h:
(WebCore::IDBObjectStoreInfo::IDBObjectStoreInfo):
(WebCore::IDBObjectStoreInfo::indexMap const):
* Source/WebCore/bindings/js/RunJavaScriptParameters.h:
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::setComposition):
* Source/WebCore/editing/Editor.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/editing/cocoa/AttributedString.mm:
(WebCore::AttributedString::AttributedString):
(WebCore::AttributedString::rangesAreSafe):
(WebCore::toNSDictionary):
(WebCore::extractDictionary):
* Source/WebCore/html/PermissionsPolicy.h:
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::setActiveContentRuleListActionPatterns):
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/ResourceLoadObserver.h:
(WebCore::ResourceLoadObserver::setDomainsWithCrossPageStorageAccess):
* Source/WebCore/page/DiagnosticLoggingClient.h:
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::updateStorageAccessUserAgentStringQuirks):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/platform/EventTrackingRegions.h:
* Source/WebCore/platform/ScreenProperties.h:
* Source/WebCore/platform/SerializedPlatformDataCueValue.h:
* Source/WebCore/platform/network/NetworkStorageSession.cpp:
(WebCore::NetworkStorageSession::setDomainsWithCrossPageStorageAccess):
(WebCore::NetworkStorageSession::grantStorageAccess):
(WebCore::NetworkStorageSession::storageAccessQuirks):
* Source/WebCore/platform/network/NetworkStorageSession.h:
* Source/WebCore/platform/network/OrganizationStorageAccessPromptQuirk.h:
(WebCore::OrganizationStorageAccessPromptQuirk::OrganizationStorageAccessPromptQuirk):
* Source/WebCore/platform/network/RFC8941.cpp:
(RFC8941::parseParameters):
(RFC8941::parseDictionary):
(RFC8941::parseDictionaryStructuredFieldValue):

ADDITIONAL FILES REDACTED FROM COMMIT MESSAGE FOR BREVITY

Canonical link: https://commits.webkit.org/285616@main
